src/main/java/core/basesyntax/serviceimpl/FruitTransactionImpl.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.db.StorageDao;
+import core.basesyntax.service.FruitTransaction;
+import core.basesyntax.strategy.FruitOperation;
+import java.util.List;
+import java.util.Map;
+
+public class FruitTransactionImpl implements FruitTransaction {
+    private static final int REPORT_HEAD_INDEX = 0;
+    private static final String DATA_DELIMITER = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int AMOUNT_INDEX = 2;
+
+    @Override
+    public void process(List<String> data, Map<String, FruitOperation> strategy,
+                        StorageDao storageDao) {
```

Comment:
please pass strategy(rename it to handlersMap) and storageDao to constructor instead - these are dependencies